packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts

```diff
@@ -0,0 +1,103 @@
+import { isNativeTokenAddress } from "../../../../constants/addresses.js";
+import type { BaseTransactionOptions } from "../../../../transaction/types.js";
+import { isNewWinningBid } from "../../__generated__/IEnglishAuctions/read/isNewWinningBid.js";
+import { bidInAuction as generatedBidInAuction } from "../../__generated__/IEnglishAuctions/write/bidInAuction.js";
+import { getAuction } from "../../read/english-auction/getAuction.js";
+import { getWinningBid } from "../../read/english-auction/getWinningBid.js";
+
+export type BidInAuctionParams = {
+  auctionId: bigint;
+} & (
+  | {
+      bidAmountWei: bigint;
+    }
+  | { bidAmount: string }
+);
+
+/**
+ * Places a bid in an English auction.
+ * @param options - The options for placing the bid.
+ * @returns A promise that resolves to the result of the bid transaction.
+ * @extension MARKETPLACE
+ * @example
+ * ```ts
+ * import { bidInAuction } from "thirdweb/extensions/marketplace";
+ *
+ * const transaction = bidInAuction({
+ *  contract,
+ *  auctionId: 0n,
+ *  bidAmount: "100",
+ * });
+ * ```
+ */
+export function bidInAuction(
+  options: BaseTransactionOptions<BidInAuctionParams>,
+) {
+  return generatedBidInAuction({
+    contract: options.contract,
+    asyncParams: async () => {
+      const [auction] = await Promise.all([
+        getAuction({
+          contract: options.contract,
+          auctionId: options.auctionId,
+        }),
+      ]);
+
+      const resolvedBidAmountWei = await (async () => {
```

Comment:
remind the reason for adding a inline function here? does it trick the bundler to respect dynamic imports or something?

Reply:
no it simply nicely encapsulates the logic needed to resolve the actual bid amount in wei, the alternative is doing something along the lines of:
```
let bidAmount;
if("bidAmount" in options){
  ...
  bidAmount = ....
}
else {
...
}
```
which just makes my skin crawl from unreadability